### PR TITLE
Fix utf8 support with MS SQL Server ODBC driver

### PR DESCRIPTION
**Pull request by lvoinea**
Hi Mark,

This pull request addresses one critical issue for us: **handling of UTF-8 statements under Linux when using the MS provided ODBC driver**. Basically, something like `insert into ΩDBC.dbo.Tαble2 ([Col Ω], [Col α]) values (N'an Ω', 'other')` would not work without this fix.

The problem can be located in several places, but `isql`, the command line tool bundled with unixODBC, is able to handle this according to our expectations. This led us to investigate a fix within `node-odbc`.

The fix is rather trivial, and follows the same approach as in `isql`. It involved simply setting the `locale` to the values configured by the user (i.e., `setlocale(LC_ALL, "");`).

Our dev environment is made up of:
- macOS 11.6 workstations
- unixODBC 2.3.11
- libiconv 1.17 (separately installed from HomeBrew)
- MS SQL Server 2017  (RTM-CU24) (KB5001228) - 14.0.3391.2 (X64)  running in a Docker container. 

Output of `locale` command:
```
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```
> NOTE: We tested the fix under Windows 10 as well and it works, although the behaviour is somewhat different. Under macOS emoji characters are not accepted, as expected (as they fall outside the UCS-2 set). However, they are accepted under Windows 10. Also worth mentioning: UTF-8 is handled by `node-odbc` under Windows 10 without the fix.

### CODE REVIEW

**Comment by kadler** [comment]
Is there no connection option for the driver to tell it which locale to use?

**Comment by lvoinea** [comment]
It looks like the MS SQL Server driver provides indeed only the wide character interfaces:

```
> nm /usr/local/lib/libmsodbcsql.18.dylib | grep SQLConnect      
000000000006dce0 T _SQLConnectW
> nm /usr/local/lib/libmsodbcsql.18.dylib | grep SQLExec         
000000000003db90 T _SQLExecDirectW
000000000003b820 T _SQLExecute
```

File: src/odbc.cpp [hunk 1]
```diff
@@ -66,6 +67,8 @@ SQLHENV ODBC::hEnv;
 
 Napi::Value ODBC::Init(Napi::Env env, Napi::Object exports) {
 
+  setlocale(LC_ALL, "");
```
[the remark this answers is not in the document]
Comment (lvoinea) [reply src/odbc.cpp:70]: Good point. However, there seems to be no out-of-the-box way of doing that in a JavaScript application on Node.js (although interest seems to be there - see https://github.com/nodejs/node/issues/28099). Or am I wrong?

**Comment by lvoinea** [comment]
>  Not sure how this would help under Linux.  I guess unixODBC would continue to assume I'm feeding in ASCII and try to convert it to UCS-2 byte by byte, which would not work.

**Comment by lvoinea** [comment]
Does this mean `node-odbc` would call de wide-character interfaces directly?

**Comment by kadler** [comment]
Sorry, yeah the solution to #292 is a slightly different problem than what you're facing. What you need is a way for node-odbc to call the wide ODBC APIs instead of the narrow ones. There is actually a way to do that, which we use on Windows, by defining UNICODE. However, this is a compile-time option and you would need to modify bindings.gyp to add this option.

Mark and I did talk about how to make this easier for an application to use this support. One way would be to provide a way for the application to specify a `use_wide` (or whatever) option on the connection to switch between the methods.

**Pull request closed without merging**